Lib/multiprocessing/pool.py

```diff
@@ -219,6 +232,23 @@ def __init__(self, processes=None, initializer=None, initargs=(),
                   self._result_handler, self._cache),
             exitpriority=15
             )
+        self._state = RUN
+
+    def __del__(self, _warn=warnings.warn):
+        if self._state == RUN:
+            _warn(f"unclosed running multiprocessing pool {self!r}", ResourceWarning,
+                  source=self)
+        elif self._state != INIT and not self._joined:
+            _warn(f"multiprocessing pool {self!r} was not joined",
```

Comment:
Should this be `stopped` instead of `joined`?

Reply:
I don't know how to describe the state here... close() or terminate() has been called. "was not joined" means that .join() should be called and it has not been called. 

"stopped" implies that there is stop() method, no? But there is no stop() method.

Reply:
Sorry, I meant `close()` instead of `stop()`

Reply:
close() doesn't ensure that all resources are released. Currently, join() must be called to ensure that all processes and threads completed.

Reply:
👍

Reply:
I thing close is more clearly, isn't?

Reply:
Read previous comments.

Reply:
As long as the user has called `terminate` or `close` (or used a context manager), a warning shouldn't be emitted.